docs/deployment-guide.md

```diff
@@ -0,0 +1,396 @@
+
+#### More on workflows
+
+So any time one of our workflows refers to “building,” it’s building a Docker image from a Dockerfile. I reiterate this because it might not be obvious at first glance. For example, here’s the build step of our `server/cloudbuild.yaml`:
+
+![alt text](doc-images/image-14.png)
+
+I guess the comment here is well-placed and kind of a giveaway, but the point is these workflow steps are pretty terse. Nowhere in the actual code do you see the words Docker or Dockerfile, and yet this step is basically a glorified `$ docker build` command that builds an image from a Dockerfile.
+
+Part of the reason this step doesn’t mention Docker is that we’re actually using Kaniko to build the image, but we’ll get to that in a moment. For now let’s go through each `key` in the build step to the left.
+
+  **name**
+    The - indicates the beginning of a distinct step within the workflow. name is used to specify a container image that will be used to run that step. Most often we’re pulling these from Artifact Registry, but not always. In this case we’re using a substitution variable ${_KANIKO_IMAGE} to hide the location of our custom Kaniko image from public view.
+
+**id**
+Used to assign a brief string identifier to a given step. It prepends this identifier to lines of the build logs but, more importantly, allows you to specify the execution order of steps when combined with the waitFor key (not pictured, but used extensively in our root-level cloudbuild.yaml).
+
+**entrypoint**
+Indicates the command to be executed upon starting a container from the image in the name key. We use bash pretty much exclusively for the purpose of executing shell scripts.
+args
+Specifies arguments to be passed either to the container’s default command or to the entrypoint explicitly declared. This is where we typically pass in paths to the shell scripts stored in the gcp directory of the GitHub repo.
+
+**env**
+Lists environment variables to be passed to the container. We use SHORT_SHA a lot: it’s a default Cloud Build substitution variable that uses a shortened form of a Git commit hash. In other words, we use it to associate a given build and deployment with a particular commit in our Git repo. We have to pass it to the container here because SHORT_SHA is not passed by default to the bash script we’re running.
+
+**secretEnv**
+Passes secrets from `Secret Manager`. This one is a bit loaded in our context, but essentially these are just more environment variables to be passed to our container and script. Not everything we pass in this field is a proper secret that needs to be encrypted: most often, in fact, it’s just a regular environment variable. More on that below.
```

Comment:
I think you have to add an extra return between the title and sentences so that a return is rendered in markdown

Reply:
I intentionally left out the return for these with bold titles, I just thought the format was a little easier to read. I did go through and add a newline between any headers and text even though it doesn't render any differently just so the markdown itself is consistent https://github.com/WonderTix/WonderTix/pull/635/commits/75cd7ffeb075c08404495a127fbdc813b08b8ca4